### ISSUE TITLE
No more than 4 jobs running in the same time

### ISSUE THREAD

**Issue by Yiadh** [ansible/awx#1158]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

 - API

##### SUMMARY
No more than 4 jobs running in the same time

##### ENVIRONMENT
* AWX version: 1.0.1
* AWX install method: openshift, minishift, docker on linux, docker for mac, boot2docker
* Ansible version:  2.4.3
* Operating System: centos7
* Web Browser: chrome

##### 
When I try to run many jobs from different jobs templates and targeting different remote hosts, I cannot get more than 4 jobs running in the same time, the other stay in the waiting queue.

**Comment by matburt** [comment]
You didn't say how much resources you have given to AWX.

AWX capacity is a thing. For AWX version 1.0.3.X I have updated the capacity algorithm a bit, see this documentation:  https://github.com/ansible/awx/blob/devel/docs/capacity.md

For 1.0.1 and current Tower releases you might be interested in a presentation I gave at Ansiblefest last year: https://www.youtube.com/watch?v=NiM4xNkauig  I go over capacity planning and algorithm in that video.

Also: http://docs.ansible.com/ansible-tower/latest/html/userguide/jobs.html#job-concurrency

**Comment by vodcav** [comment]
if you want to run more than 4 jobs in the same time you should have more awx_task container. only 4 jobs can be executed concurrently per a awx_task container. It it very inconvenient.

**Comment by matburt** [comment]
> only 4 jobs can be executed concurrently per a awx_task container.

This is patently false do not listen to it. The maximum number of simultaneous jobs is based on the capacity of your system, see the documentation on this system here: https://github.com/ansible/awx/blob/devel/docs/capacity.md

**Comment by icsm2017** [comment]
I agree, but how do you explain that despite the ram update of 4 to 16GB, AWX still only launch 4 jobs simultaneously ?

I tested the same template (with same project, playbook, inventory) on the trial version of tower and I actually went ~20 jobs simultaneously on the same server.

Would there be an extra parameter to set for AWX ?

Environment : 
AWX version: 1.0.2.0
AWX install method: docker on linux
Ansible version: 2.4.3
Operating System: CentOS 7/RH7 for Tower
Web Browser: chrome

**Comment by matburt** [comment]
This is a fairly old version of AWX... can you try 1.0.4.X?

**Comment by icsm2017** [comment]
I try with version 1.0.4.48 and I have exactly the same problem. I try also with more memory and CPU (24GB/4CPU), always stucks at 4 jobs simultaneous and the others jobs still pending.

Environment :
AWX version: 1.0.4.48
AWX install method: docker on linux
Ansible version: 2.4.3
Operating System: Red Hat 7.4
Web Browser: chrome

**Comment by rahniwalden** [comment]
Why was this closed? This is still a problem and I'm running the latest version of AWS (6.1.0.0)

**Comment by matburt** [comment]
We haven't seen any information that *actually* demonstrates a problem or point towards a root cause.

**Comment by rahniwalden** [comment]
@matburt what information can I give you? I have AWX running in openshift and we're seeing this same issue. Someone stated above this is due to compute constraints but we're running in OpenShift and the container sees the following:

16 CPU
62 GB Mem

I presume these are from the host machine's compute, but appear to still be available to the container. I checked those job settings in AWX and saw thing especially telling...

I tried setting: "SYSTEM_TASK_FORKS_CPU": "10"
but that did not seem to affect anything. 

It could be im just missing some setting or value, but we can't seem to get more than 4 jobs to run simultaneously.. So not sure what we're missing here, or doing wrong.
